sentinel-adapter/sentinel-jdbc-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/jdbc/SentinelJDBCStatement.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 1999-2022 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.csp.sentinel.adapter.jdbc;
+
+import com.alibaba.csp.sentinel.adapter.jdbc.delegate.StatementDelegate;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * @author icodening
+ * @date 2022.02.09
+ */
+public class SentinelJDBCStatement extends StatementDelegate {
+
+    public SentinelJDBCStatement(Statement delegate) {
+        super(delegate);
+    }
+
+    @Override
+    public ResultSet executeQuery(String sql) throws SQLException {
+        setSQL(sql);
+        return sentinelEntryInternal(() -> super.executeQuery(sql), true);
```

Comment:
For normal JDBC statement, it might bring a large number of different SQLs, which generates too much Sentinel "resource" and cause huge memory footprint. Maybe we need some mechanism to avoid it.

Reply:
> For normal JDBC statement, it might bring a large number of different SQLs, which generates too much Sentinel "resource" and cause huge memory footprint. Maybe we need some mechanism to avoid it.

I have to think about this problem, maybe we can add a custom resource method, and use SQL when the custom resource cannot be obtained. example:
````java
protected String getResourceName() {
    String currentResourceName = JDBCResourceUtil.getCurrentThreadJDBCResourceName();
    if (StringUtil.isNotBlank(currentResourceName)) {
        return currentResourceName;
    }
    return getSQL();
}
````
However, when doing this, it is also necessary to adapt JPA and MyBatis, the "resource" needs to use repoInterface.methodName/namespace+id(in MyBatis), such as: com.alibaba.repository.UserRepository.saveUser. MyBatis interceptor example:
````java
public Object intercept(Invocation invocation) throws Throwable {
    MappedStatement ms = getMappedStatement(invocation);
    String resource = ms.getId(); //namespace+sqlName
    JDBCResourceUtil.setCurrentThreadJDBCResourceName(resource);
    return invocation.proceed();
}
````
How about this way?